src/net/l0/net_rx.c

```diff
@@ -33,14 +35,7 @@ int net_rx(struct sk_buff *skb) {
 		return 0; /* error: invalid size */
 	}
 
-	/* parse L2 header */
-	assert(skb->dev->ops != NULL);
-	assert(skb->dev->ops->parse_hdr != NULL);
-	if (0 != skb->dev->ops->parse_hdr(skb, &hdr_info)) {
-		log_error("%p can't parse header", skb);
-		skb_free(skb);
-		return 0; /* error: can't parse L2 header */
-	}
```

Comment:
You removed this code, but src/net/l2/ethernet.c:68 still remains:
```
const struct net_device_ops ethernet_ops = {
    .build_hdr = ethernet_build_hdr,
    .parse_hdr = ethernet_parse_hdr,
    .check_addr = is_valid_ether_addr,
    .check_mtu = ethernet_check_mtu
};
```
And now it is unused. We do not need this logic anymore? And now, all our network stack depends on ethernet here because of `eth_hdr(skb)->h_proto`, is it ok?

Reply:
I mean we can completely delete parse_hdr if you are using eth_hdr directly now.

Reply:
Yes, I agree, Done